tests/testthat/test-list-combine.R

```diff
@@ -0,0 +1,37 @@
+test_that("basic `list_combine()` works", {
```

Comment:
Just a few `list_combine()` specific tests. Most new tests are for `list_unchop()`.

When we move to "officially" exporting and documenting `list_combine()`, I'll copy all tests over from `list_unchop()` and "freeze" the `list_unchop()` set of tests.